pkg/controller/plan/adapter/openstack/builder.go

```diff
@@ -420,42 +420,61 @@ func (r *Builder) mapResources(vm *model.Workload, object *cnv.VirtualMachineSpe
 	cpuCores = 1
 	cpuThreads = 1
 
-	if flavorCPUSockets, ok := vm.Flavor.ExtraSpecs[FlavorCpuSockets]; ok {
-		if sockets, err := strconv.Atoi(flavorCPUSockets); err == nil {
-			cpuSockets = sockets
+	if imageCPUSockets, ok := vm.Image.Properties[CpuSockets]; ok {
+		if socketsStr, ok := imageCPUSockets.(string); ok {
+			if sockets, err := strconv.Atoi(socketsStr); err == nil {
+				cpuSockets = sockets
+			} else {
+				r.Log.Error(err, "unable to parse image property", "property", CpuSockets)
```

Comment:
should it be something like 
```suggestion
				r.Log.Error(err, "unable to parse image property", "cpuSockets", CpuSockets)
```
here and below?

Reply:
I don't think so, `CpuSockets` value is `hw_cpu_sockets` which is the image property name, not the value.

Reply:
Yes, the value is in `vm.Image.Properties[CpuSockets]`. A more likely alternative is to have: `..., "property", CpuSockets, "value", vm.Image.Properties[CpuSockets]`.

Reply:
> Yes, the value is in `vm.Image.Properties[CpuSockets]`. A more likely alternative is to have: `..., "property", CpuSockets, "value", vm.Image.Properties[CpuSockets]`.

yeah, logging the value can be useful

Reply:
done